Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -1450,24 +1454,57 @@ PhysicalParticleContainer::AddPlasmaFlux (PlasmaInjector const& plasma_injector,
         if (refine_injection) {
             fine_overlap_box = overlap_box & amrex::shift(fine_injection_box, -shifted);
         }
+
+#ifdef AMREX_USE_EB
+        // Extract data structures for embedded boundaries
+        amrex::Array4<const typename FabArray<EBCellFlagFab>::value_type> eb_flag_arr;
+        amrex::Array4<const amrex::Real> eb_bnd_area_arr;
+        amrex::Array4<const amrex::Real> eb_bnd_normal_arr;
+        amrex::Array4<const amrex::Real> eb_bnd_cent_arr;
+        if (inject_from_eb) {
+            eb_flag_arr = eb_flag->array(mfi);
+            eb_bnd_area_arr = eb_bnd_area->array(mfi);
+            eb_bnd_normal_arr = eb_bnd_normal->array(mfi);
+            eb_bnd_cent_arr = eb_bnd_cent->array(mfi);
+        }
+#endif
+
         amrex::ParallelForRNG(overlap_box, [=] AMREX_GPU_DEVICE (int i, int j, int k, amrex::RandomEngine const& engine) noexcept
         {
             const IntVect iv(AMREX_D_DECL(i, j, k));
             amrex::ignore_unused(j,k);
 
-            auto lo = getCellCoords(overlap_corner, dx, {0._rt, 0._rt, 0._rt}, iv);
-            auto hi = getCellCoords(overlap_corner, dx, {1._rt, 1._rt, 1._rt}, iv);
-
-            if (flux_pos->overlapsWith(lo, hi))
+            // Determine the number of macroparticles to inject in this cell (num_ppc_int)
+#ifdef AMREX_USE_EB
+            amrex::Real num_ppc_real_in_this_cell = num_ppc_real; // user input: number of macroparticles per cell
+            if (inject_from_eb) {
+                // Injection from EB
+                // Skip cells that are not partially covered by the EB
+                if (eb_flag_arr(i,j,k).isRegular() || eb_flag_arr(i,j,k).isCovered()) { return; }
+                // Scale by the (normalized) area of the EB surface in this cell
+                num_ppc_real_in_this_cell *= eb_bnd_area_arr(i,j,k);
+            } else
+#else
+            amrex::Real const num_ppc_real_in_this_cell = num_ppc_real; // user input: number of macroparticles per cell
+#endif
             {
-                auto index = overlap_box.index(iv);
-                int r = 1;
-                if (fine_overlap_box.ok() && fine_overlap_box.contains(iv)) {
-                    r = compute_area_weights(rrfac, flux_normal_axis);
-                }
-                const int num_ppc_int = static_cast<int>(num_ppc_real*r + amrex::Random(engine));
-                pcounts[index] = num_ppc_int;
+                // Injection from a plane
+                auto lo = getCellCoords(overlap_corner, dx, {0._rt, 0._rt, 0._rt}, iv);
+                auto hi = getCellCoords(overlap_corner, dx, {1._rt, 1._rt, 1._rt}, iv);
+                // Skip cells that do not overlap with the plane
+                if (!flux_pos->overlapsWith(lo, hi)) { return; }
+            }
+            const int num_ppc_int = static_cast<int>(num_ppc_real_in_this_cell + amrex::Random(engine));
```

Comment:
This looses the change made in PR #5272. The calculation of `num_ppc_int` should happen after the `r` factor is included.